### ISSUE TITLE
AlexNet 无法 train，loss 初始 6.91，无改变

### ISSUE THREAD

**Issue by admk** [Aaron-Zhao123/mayo-dev#1]
Export 一个 yaml：
```yaml
---
dataset:
    background_class: {has: true, use: false}
    channel_means: [0.485019607843137, 0.4579607843, 0.407607843137255]
    name: imagenet
    num_classes: 1001
    num_examples_per_epoch: {train: 1281167, validate: 50000}
    path: {train: imagenet/train-*, validate: imagenet/validation-*}
    shape: {channels: 3, height: 224, width: 224}
hyper:
-   biases_initializer: &id001 {type: tensorflow.constant_initializer, value: 0.01}
    weights_initializer: &id002 {type: tensorflow.glorot_normal_initializer}
-   biases_initializer: *id001
    padding: same
    type: convolution
    weights_initializer: *id002
-   biases_initializer: *id001
    type: fully_connected
    weights_initializer: *id002
- {alpha: 2.0e-05, beta: 0.75, bias: 1.0, depth_radius: 2, type: local_response_normalization}
logits: logits
name: alexnet
net:
-   biases_initializer: *id001
    kernel_size: 11
    name: conv1
    num_outputs: 64
    padding: valid
    stride: 4
    type: convolution
    weights_initializer: *id002
- {alpha: 2.0e-05, beta: 0.75, bias: 1.0, depth_radius: 2, name: norm1,
    type: local_response_normalization}
- {kernel_size: 3, name: pool1, padding: valid, stride: 2, type: max_pool}
-   biases_initializer: *id001
    kernel_size: 5
    name: conv2
    num_outputs: 192
    padding: same
    stride: 1
    type: convolution
    weights_initializer: *id002
- {alpha: 2.0e-05, beta: 0.75, bias: 1.0, depth_radius: 2, name: norm2,
    type: local_response_normalization}
- {kernel_size: 3, name: pool2, padding: valid, stride: 2, type: max_pool}
-   biases_initializer: *id001
    kernel_size: 3
    name: conv3
    num_outputs: 384
    padding: same
    stride: 1
    type: convolution
    weights_initializer: *id002
-   biases_initializer: *id001
    kernel_size: 3
    name: conv4
    num_outputs: 384
    padding: same
    stride: 1
    type: convolution
    weights_initializer: *id002
-   biases_initializer: *id001
    kernel_size: 3
    name: conv5
    num_outputs: 256
    padding: same
    stride: 1
    type: convolution
    weights_initializer: *id002
- {kernel_size: 3, name: pool5, padding: valid, stride: 2, type: max_pool}
- {name: flatten5, type: flatten}
-   biases_initializer: *id001
    name: fc6
    num_outputs: 4096
    type: fully_connected
    weights_initializer: *id002
- {keep_prob: 0.5, name: dropout6, type: dropout}
-   biases_initializer: *id001
    name: fc7
    num_outputs: 4096
    type: fully_connected
    weights_initializer: *id002
- {keep_prob: 0.5, name: dropout7, type: dropout}
-   activation_fn: null
    biases_initializer: *id001
    name: logits
    num_outputs: num_classes
    type: fully_connected
    weights_initializer: *id002
preprocess:
    final:
    - {scale: 2.0, shift: -1.0, type: linear_map}
    train:
    - {type: distort_bbox}
    - {type: distort_color}
    - {type: random_flip}
    validate: []
system:
    batch_size: 256
    checkpoint: {load: false, save: true}
    log_level: {mayo: info, tensorflow: 2}
    max_steps: 10000000
    num_gpus: 1
    num_preprocess_threads: 4
    search_paths:
        checkpoints: [checkpoints]
        datasets: ['D:\Downloads']
        summaries: [summaries]
    use_pdb: true
train:
    learning_rate: {_num_epochs_per_decay: 30.0, decay_rate: 0.16, decay_steps: 150136.7578125,
        learning_rate: 0.001, staircase: true, type: tensorflow.train.exponential_decay}
    moving_average_decay: 0.9999
    optimizer: {decay: 0.9, epsilon: 1.0, momentum: 0.9, type: tensorflow.train.RMSPropOptimizer}
```
怀疑 preprocessing 有问题。

**Comment by Aaron-Zhao123** [comment]
试了去除 subtract image mean, 依旧卡在6.9

**Comment by admk** [comment]
参考讨论：https://github.com/BVLC/caffe/issues/401
参考 hyperparameter：https://gist.github.com/jimmie33/0585ed9428dc5222981f#file-solver-prototxt

**Comment by Aaron-Zhao123** [comment]
参考 tf.slim 实现：
https://github.com/tensorflow/models/tree/master/slim

**Comment by Aaron-Zhao123** [comment]
Running Alexnet:
⠋ epoch: 6.49 | loss: 4.97    ±  5% | acc:   9.9% | ckpt: 5.99 | tp:  135/s

Variance got better when loss decreases further

**Comment by Aaron-Zhao123** [comment]
checkpoints/alexnet/imagenet/checkpoint-44999
-       top1: 17.35%, top5: 37.93% [50000 images]

**Comment by Aaron-Zhao123** [comment]
⠏ epoch: 32.79 | loss: 3.46    ±  3% | acc:  29.3% | ckpt: 31.97 | tp:  191/s

**Comment by Aaron-Zhao123** [comment]
⠇ epoch: 47.53 | loss: 3.36    ±  4% | acc:  30.8% | ckpt: 46.96 | tp:  197/s

**Comment by Aaron-Zhao123** [comment]
⠴ epoch: 59.39 | loss: 3.27    ±  4% | acc:  32.0% | ckpt: 58.95 | tp:  189/s

**Comment by admk** [comment]
A pre-trained AlexNet v2 model (no groups) can be found at: https://github.com/pytorch/vision/blob/master/torchvision/models/alexnet.py

Using pytorch, the tensors of the pre-trained model can be automatically downloaded and unpickled by invoking:
```python
model_zoo.load_url('https://download.pytorch.org/models/alexnet-owt-4df8aa71.pth')
```
